lib/Doctrine/MongoDB/Aggregation/Stage/AbstractBucket.php

```diff
@@ -75,4 +75,11 @@ protected function convertExpression($expression)
      * @return array
      */
     abstract protected function getExtraPipelineFields();
+
+    /**
+     * Returns the stage name with the dollar prefix
+     *
+     * @return string
+     */
+    abstract protected function getStageName();
```

Comment:
AbstractBucket is annotated with `@internal` so I wouldn't even consider this a BC break.

Reply:
Oh, I missed that completely. Even better! 👍